File: two_balls_python_kolpakov/basket.py
from random import randrange

class Basket:
    def __init__(self):
        self.balls = []
        self.balls.append('white')

        if randrange(2):
            self.balls.append('white')
        else:
            self.balls.append('black')

    def getBall(self):
        if len(self.balls) == 1:
            return self.balls.pop()
        elif len(self.balls) == 2:
            return self.balls.pop(randrange(2))
        else:
            return False

File: three_balls_ruby_kolpakov/basket.rb
class Basket

    def initialize
        @balls = []
        @balls << WHITE

        second_rand = rand(1..5)

        if second_rand == 1
            @balls << BLACK
        else
            @balls << WHITE
        end

        third_rand = rand(1..5)

        if third_rand == 1
            @balls << WHITE
        elsif third_rand <= 3
            @balls << BLACK
        else
            @balls << RED
        end
    end

    def take_ball
        random = rand(0..(@balls.length - 1))
        ball = @balls[random]
        @balls.delete_at(random)

        return ball
    end

    def get_balls
        return @balls
    end

end
